crates/sui-bridge/src/orchestrator.rs

```diff
@@ -130,10 +137,27 @@ where
                 info!("Observed Sui bridge event: {:?}", bridge_event);
 
                 if let Some(action) = bridge_event
+                    .clone()
```

Comment:
you don't need to clone the event if you handle NewTokenEvent first, also if the event is a NewTokenEvent, we can skip try it into action.